### ISSUE TITLE
Sure, but what's the point of all this if you can't make money?

### ISSUE THREAD

**Issue by nvhai245** [zakirullin/cognitive-load#33]
Every developer is a philosopher these days. Do you think management care about these stuff? Being a senior dev today means that you weren't promoted into management position and stuck at your desk coding.  That's it. Absorb more of this shit and you stay poor forever. All this leetcode races, algorithms, clean code... doesn't mean shit anymore. Devin would do better than you all someday giving it enough time.

Oh and you probably got paid less that your scrum master who sit all day doing nothing. While you struggling with your "cognitive load".

**Comment by mikhail-barg** [comment]
> Being a senior dev today means that you weren't promoted into management position and stuck at your desk coding

This is not true. There are "expert" career tracks. No reason to morf from a good programmer into a poor/mediocre manager.

**Comment by ucrt0** [comment]
> Absorb more of this shit and you stay poor forever.

I think this not about being paid or not, rather a responsible way to make code reasonable and preserving it for our future self.

**Comment by turjachaudhuri-opensource** [comment]
I believe the idea of the article is to explain that Cognitive Load is real, and it impacts productivity of everyone involved.
So, we should always keep an eye out for reducing the same, wherever possible.

Regarding making money, etc, technically speaking you ca write a book about Cognitive load, and sell it and make some money.
But, that's beyond the point - the idea would be have a sustainable framework that allows us to be productive quickly.

**Comment by antz-H** [comment]
> I believe the idea of the article is to explain that Cognitive Load is real, and it impacts productivity of everyone involved. So, we should always keep an eye out for reducing the same, wherever possible.
> 

I agree with you . And I think author just disscuss the Cognitive Load from the coding view, but the idea also is suitable for more things ,such as writing , creating etc.

**Comment by devashishraj** [comment]
> Every developer is a philosopher these days. Do you think management care about these stuff? Being a senior dev today means that you weren't promoted into management position and stuck at your desk coding. That's it. Absorb more of this shit and you stay poor forever. All this leetcode races, algorithms, clean code... doesn't mean shit anymore. Devin would do better than you all someday giving it enough time.
> 
> Oh and you probably got paid less that your scrum master who sit all day doing nothing. While you struggling with your "cognitive load".

@nvhai245 you remind me of this : https://youtu.be/L0zVJmsz5ww?si=2xYOGcs590TLUrzq&t=254

**Comment by johnnyshields** [comment]
Porque no los dos? If you are a founder or dev at a lean company, adopting habitual practices which reduce cognitive overhead in a code base makes it much easy to build/maintain over a long period, and hire/collaborate with a team as you scale. And these same practices even can be applied overtime to a messy legacy codebase (best to start by writing lots of tests first!)

If you are a stuck at a depressing coding deskjob at Big Tech making megabucks then sure, life sucks, your soul is too crushed to even enjoy your money, and why bother trying anything good/sustainable because management will kill it instantly. Point taken.

**Comment by jdmarshall** [comment]
Pretty sure 'philosophical' is a job requirement for a Principal Engineer and a nice-to-have for Staff.

Younger me was at least partially guilty of the aspersion you're casting. But the problem is that keeping a project moving gets harder and harder and there's less incentive for your good people to stay. So there's a line beyond which design and DX is white tower bullshit, but there's a certain serving size that is necessary to keep a project from bogging down irrevocably. 

And the 'irrevocable' point is only certain when it's in the rear view mirror. It's when that one guy left, and he had one foot out the door for four months before he finally left, so the point of no return was actually six months ago, at the latest.

Anyone can make a version 1 product. It takes discipline to get to version 2.
